newrelic-weaver/src/main/java/com/newrelic/weave/utils/ReturnInsnProcessor.java

```
@@ -0,0 +1,103 @@
+package com.newrelic.weave.utils;
+
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.*;
+import org.objectweb.asm.tree.analysis.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ReturnInsnProcessor {
+
+    private static final int EXPECTED_RETURN_STACK_SIZE = 1;
+
+    /***
+     * Clear the stacks at every A and I-type return instruction of a method, to their EXPECTED_RETURN_STACK_SIZE of 1.
+     * This mutates the given MethodNode and is not thread safe.
+     *
+     * @param owner The owning class of the method
+     * @param mn MethodNode representing the method to be modified
+     */
+    public static void clearReturnStacks(String owner, MethodNode mn) {
+        Map<AbstractInsnNode, Integer> stacks;
+        try {
+            stacks = getReturnStacks(owner, mn);
+        } catch (AnalyzerException e) {
+            //Something went wrong calculating return stacks, leave the method unmodified.
```

This is a silent failure